rust/yarp-sys/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+use std::ffi::c_char;
+
+// In `build.rs`, we use `bindgen` to generate bindings based on C headers and `librubyparser`. Here
+// is where we pull in those bindings and make them part of our library.
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+pub type yp_parser_t_lex_modes = yp_parser__bindgen_ty_1;
+
+//  ╭──────────────────────────────────────────────────────────────────────────────╮
+//  │ Hand-rolling `yp_string_t` things                                            │
+//  │                                                                              │
+//  │ `bindgen` can generate these, but because of the nested `union`s, the        │
+//  │ types end up with weird names. I'm sure there's a way to fix that, but I     │
+//  │ haven't been able to. So I'm hand-rolling these for now.                     │
+//  ╰──────────────────────────────────────────────────────────────────────────────╯
+#[repr(C)]
+#[derive(Clone, Copy)]
```

Comment:
can this type derive `Default` to match the bindgen settings?

Reply:
It can, yeah. My only hesitancy is what the default variant should be for `yp_string_t_type`; right now it's `YP_STRING_SHARED`, but does that make sense? I'd almost rather hold off and add this in later after the lib has gotten some use out of it... unless someone familiar with the C API can weigh in as to what's the most common variant of string used?